packages/ingest/test/test-S3ProviderClient.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const fs = require('fs');
+const { basename } = require('path');
+const test = require('ava');
+const S3 = require('@cumulus/aws-client/S3');
+const { randomString } = require('@cumulus/common/test-utils');
+const S3ProviderClient = require('../S3ProviderClient');
+
+const localPath = './tmp.json';
+
+test.before(async (t) => {
+  t.context.sourceBucket = randomString();
+  t.context.sourcePrefix = randomString();
+  t.context.sourceKey = `${t.context.sourcePrefix}/${randomString()}`;
+  t.context.targetBucket = randomString();
+  t.context.fileContent = JSON.stringify({ type: 'fake-test-object' });
+
+  await Promise.all([
+    S3.createBucket(t.context.sourceBucket),
+    S3.createBucket(t.context.targetBucket)
+  ]);
+
+  await S3.s3PutObject({
+    Bucket: t.context.sourceBucket,
+    Key: t.context.sourceKey,
+    Body: t.context.fileContent
+  });
+});
+
+test.after.always(async (t) => {
+  fs.unlinkSync(localPath);
+  return Promise.all([
+    S3.recursivelyDeleteS3Bucket(t.context.sourceBucket),
+    S3.recursivelyDeleteS3Bucket(t.context.targetBucket)
+  ]);
+});
+
+test('S3ProviderClient constructor throws error if no bucket specified', (t) => {
+  t.throws(
+    () => new S3ProviderClient(),
+    'bucket is required'
+  );
+  t.throws(
+    () => new S3ProviderClient({}),
+    'bucket is required'
+  );
+});
+
+test.serial('S3ProviderClient.list lists objects under a path in a bucket', async (t) => {
+  const s3ProviderClient = new S3ProviderClient({ bucket: t.context.sourceBucket });
+
+  const files = await s3ProviderClient.list(t.context.sourcePrefix);
+  t.is(files.length, 1);
+  t.is(files[0].name, basename(t.context.sourceKey));
+});
+
+test.serial('S3ProviderClient.downloads downloads a file to local disk', async (t) => {
+  const s3ProviderClient = new S3ProviderClient({ bucket: t.context.sourceBucket });
+
+  await s3ProviderClient.download(t.context.sourceKey, localPath);
+  t.true(fs.existsSync(localPath));
+  t.is(fs.readFileSync(localPath).toString(), t.context.fileContent);
+});
+
+test.serial('S3ProviderClient.sync syncs a file to a target S3 location', async (t) => {
+  const s3ProviderClient = new S3ProviderClient({ bucket: t.context.sourceBucket });
+  const targetKey = 'target.json';
+
+  await s3ProviderClient.sync(t.context.sourceKey, t.context.targetBucket, targetKey);
+  t.is(await S3.getTextObject(t.context.targetBucket, targetKey), t.context.fileContent);
+});
+
+test.serial('S3ProviderClient.sync throws an error if the source file does not exist', async (t) => {
+  const s3ProviderClient = new S3ProviderClient({ bucket: t.context.sourceBucket });
+
+  await t.throwsAsync(
+    () => s3ProviderClient.sync('non-existent', t.context.targetBucket, 'target.json'),
```

Comment:
There's nothing wrong with this - I'm just still not clear when throwAsync requires you to put your async function in an anonymous callback

Reply:
I could alternatively pass in `s3ProviderClient.sync.bind(null, 'non-existent', t.context.targetBucket, 'target.json');`. The point is to have a function that `throwsAsync` can call.

Reply:
Yeah , I don't think this should change. I just get confused bc in some cases our `throwsAsync` calls pass in an async function directly, in other cases we wrap it in a function

For example:

https://github.com/nasa/cumulus/blob/master/packages/api/tests/lambdas/test-verifyProviderSecretsMigration.js#L139

Reply:
I see. So if you look at https://github.com/avajs/ava/blob/master/docs/03-assertions.md#throwsasyncthrower-expectation-message you'll see that `throwsAsync` accepts *either* of a function, *or* a promise.

Since calling an async function without `await` returns a promise, either way works.